geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/blocks/BlockTracking.kt

```diff
@@ -0,0 +1,95 @@
+package com.mineinabyss.geary.papermc.tracking.blocks
+
+import com.mineinabyss.geary.addons.dsl.GearyAddonWithDefault
+import com.mineinabyss.geary.modules.geary
+import com.mineinabyss.geary.papermc.tracking.blocks.helpers.GearyBlockPrefabQuery
+import com.mineinabyss.geary.papermc.tracking.blocks.systems.TrackOnSetBlockComponent
+import com.mineinabyss.geary.prefabs.PrefabKey
+import com.mineinabyss.idofront.di.DI
+import org.bukkit.Instrument
+import org.bukkit.Material
+import org.bukkit.Note
+import org.bukkit.block.BlockFace
+import org.bukkit.block.data.BlockData
+import org.bukkit.block.data.type.CaveVines
+import org.bukkit.block.data.type.NoteBlock
+import org.bukkit.block.data.type.Tripwire
+
+val gearyBlocks by DI.observe<BlockTracking>()
+interface BlockTracking {
+    val blockMap: Map<BlockData, Int>
+    val blockProvider: GearyBlockProvider
+    val block2Prefab: Block2Prefab
+    val blockPrefabs: GearyBlockPrefabQuery
+
+    fun createBlockData(prefabKey: PrefabKey): BlockData? = blockProvider.serializePrefabToBlockData(prefabKey)
+
+    companion object : GearyAddonWithDefault<BlockTracking> {
+        override fun default(): BlockTracking = object : BlockTracking {
+            override val blockMap = createBlockMap()
+            override val blockProvider = GearyBlockProvider()
+            override val blockPrefabs = GearyBlockPrefabQuery()
+            override val block2Prefab = Block2Prefab()
+        }
+
+        override fun BlockTracking.install() {
+            geary.pipeline.addSystems(
+                TrackOnSetBlockComponent()
```

Comment:
GearyListener to fill the prefabMap whenever an entity has its PrefabKey and SetBlock components set